libsrc/common/_time_t_to_tm.s

```diff
@@ -0,0 +1,121 @@
+;
+; Colin Leroy-Mira, 2024
+;
+; struct tm* __fastcall__ _time_t_to_tm (const time_t t)
+;
+; Helper to gmtime and localtime. Breaks down a number of
+; seconds since Jan 1, 1970 into days, hours and seconds,
+; so that each of them fits in 16 bits; passes the
+; result to _mktime which fixes all values in the struct,
+; and returns a pointer to the struct to callers.
+;
+
+        .export         __time_t_to_tm
+        .import         udiv32, _mktime
+        .importzp       sreg, tmp3, ptr1, ptr2, ptr3, ptr4
+
+        .include        "time.inc"
+
+__time_t_to_tm:
+        ; Divide number of seconds since epoch, in ptr1:sreg,
+        ; by 86400 to get the number of days since epoch, and
+        ; the number of seconds today in the remainder.
+
+        ; Load t as dividend (sreg is already set by the caller)
+        sta     ptr1
+        stx     ptr1+1
+
+        ; Load 86400 as divisor
+        lda     #$80
+        sta     ptr3
+        lda     #$51
+        sta     ptr3+1
+        lda     #$01
+        sta     ptr4
+        lda     #$00
+        sta     ptr4+1
+
+        ; Clear TM buf while we have zero in A
+        ldx     #.sizeof(tm)-1
+:       sta     TM,x
+        dex
+        bne     :-
+
```

Comment:
I assume you mean to clear all of `TM` here, in which case you'll need `bpl`, not `bne` here.

Reply:
Indeed!